pyrit/score/scorer.py

```diff
@@ -16,36 +17,39 @@ class Scorer(abc.ABC):
     scorer_type: ScoreType
 
     @abstractmethod
-    async def score_async(self, request_response: PromptRequestPiece) -> list[Score]:
+    async def score_async(self, request_response: PromptRequestPiece, *, task: Optional[str] = None) -> list[Score]:
         """
         Score the request_response, add the results to the database
         and return a list of Score objects.
 
         Args:
             request_response (PromptRequestPiece): The request response to be scored.
+            task (str): The task based on which the text should be scored.
 
         Returns:
             list[Score]: A list of Score objects representing the results.
         """
         raise NotImplementedError("score_async method not implemented")
 
     @abstractmethod
-    def validate(self, request_response: PromptRequestPiece):
+    def validate(self, request_response: PromptRequestPiece, *, task: Optional[str] = None):
```

Comment:
Same suggestion as below.

```suggestion
        def validate(self, *, request_response: PromptRequestPiece, task: Optional[str] = None):
```

Reply:
see https://github.com/Azure/PyRIT/pull/274#discussion_r1687244633